src/utils/scroll.js

```diff
@@ -0,0 +1,179 @@
+
+const mouse = require("mouse-pos");
+const jbone = require("jbone");
+
+import { get, noop } from "lodash";
+
+class ScrollBody {
+  constructor(g, backboneView, onScrollCallback) {
+    this.g = g;
+    this.backboneView = backboneView;
+    this.onScrollCallback = () => ((onScrollCallback || noop).call(this.backboneView, arguments));
+
+    this.dragStart = [];
+    this.dragStartScroll = [];
+  }
+
+  getScrollEvents() {
+    const events = {};
+
+    // add scroll support
+    events.mousewheel = this._onScroll;
+    events.DOMMouseScroll = this._onScroll;
+    events.wheel = this._onScroll;
+
+    // add touch/drag scroll support
+    events.mousedown = this._onmousedown;
+    events.touchstart = this._ontouchstart;
+
+    // call the callback function for scrolls whenever scroll offsets change
+    this.backboneView.listenTo(this.g.zoomer, "change:_alignmentScrollLeft change:_alignmentScrollTop", this.onScrollCallback);
+
+    return events;
+  }
+
+  /**
+   * @param {*} e 
+   * @returns 
+   */
+  _onScroll = (e) => {
+    const delta = mouse.wheelDelta(e);
+    const scrollCorrected = this._checkScrolling([
+      this.g.zoomer.get('_alignmentScrollLeft') + delta[0],
+      this.g.zoomer.get('_alignmentScrollTop') + delta[1]
+    ]);
+    this.g.zoomer.set('_alignmentScrollLeft', scrollCorrected[0]);
+    this.g.zoomer.set('_alignmentScrollTop', scrollCorrected[1]);
+    e.preventDefault();
+    e.stopPropagation();
+  }
+  
+  // converts touches into old mouse event
+  _ontouchmove = (e) => {
+    this._onmousedrag(e.changedTouches[0], true);
+    e.preventDefault();
+    e.stopPropagation();
+  }
+
+  // start the dragging mode
+  _onmousedown = (e) => {
+    this.dragStart = mouse.abs(e.originalEvent);
```

Comment:
This was previously using the backbone event `e`, but I've changed it to use `e.originalEvent`.

This fixes [SS-44989](https://jira.schrodinger.com/browse/SS-44989).
The bug occurs because of how jbone wraps native events 😐
See https://github.com/kupriyanenko/jbone/blob/80b99258ef0fcd23f9d7451fbe4d87e73ded94f1/src/event.js#L1-L26 

It's creating a synthetic BoneEvent and for whatever reason they've decided to NOT copy over event properties that are "falsey".
The problem -- especially for mouse events -- is that a lot of values (eg: `clientX`) don't get copied when it's zero.

So if the user does a scroll really slowly to the left, and the cursor goes at the beginning of the document, then the offset becomes zero... 
(For iframes this is basically the moment the cursor exits the iframe..)

Thus the offset property is NOT present anymore in the viewer's scroll event.
Later the viewer's offset calculation becomes `NaN` down the line cause arithematic operations over undefined prpoerties often go like `undefined - 50 === NaN` (thanks JavaScript!), and this breaks the viewer completely 🤦.